### PR TITLE
docs: restructure deploy doc

### PR DESCRIPTION
**Pull request by skoh7645**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Restructure "deploying" doc to be clearer

## Related Issues
Related to #359 
Fixes: https://github.com/appsody/appsody/issues/885

**Note**: the `### Deploying your application through Docker Hub` and `### Deploying your application to a custom registry` sections get fixed as part of https://github.com/appsody/website/pull/566

### CODE REVIEW

File: content/docs/faq.md [hunk 1]
```diff
@@ -159,4 +159,10 @@ The `project.yaml` file is validated when a new Appsody project is initialized.
 * moved to a different directory
 * are being initialized into the same directory as a previous project
 
-Their entry is removed from the `project.yaml` file and the associated volumes are deleted. This validation ensures that volumes exist for current projects only and that volumes are not reused between different projects of the same name.
\ No newline at end of file
+Their entry is removed from the `project.yaml` file and the associated volumes are deleted. This validation ensures that volumes exist for current projects only and that volumes are not reused between different projects of the same name.
+
+## 12. How do I deploy my application to a local Minikube cluster?
+
+Currently, Appsody only supports using a local Docker daemon. If you use the Minikube docker daemon, issuing `eval $(minikube docker-env)` , [appsody extract](docs/cli-commands/#appsody-extract) and [appsody build](docs/cli-commands/#appsody-build) commands will not work as expected.
```
Comment (helenmasters) [inline content/docs/faq.md:166]: Reworded so only is in the right place... "Currently, Appsody supports you using a local Docker daemon only."

Comment (helenmasters) [inline content/docs/faq.md:166]: docker daemon -> Docker daemon

File: content/docs/using-appsody/deploying.md [hunk 2]
```diff
@@ -9,55 +9,59 @@ The Appsody CLI provides the [appsody deploy](/docs/cli-commands/#appsody-deploy
 
 You can delegate the build and deployment steps to an external pipeline, such as a Tekton pipeline that consumes the source code of your Appsody project after you push it to a GitHub repository. Within the pipeline, you can run [appsody build](/docs/using-appsody/building), which builds the application image and generates a deployment manifest. You can use the manifest to deploy your application to a Kubernetes environment where the Appsody operator is installed.
 
+> Currently we only support a local Docker daemon. To workaround this, see the [FAQ](/docs/faq/#12-how-do-i-deploy-my-application-to-a-local-minikube-cluster).
```
Comment (helenmasters) [inline content/docs/using-appsody/deploying.md:12]: "Currently, Appsody supports you using a local Docker daemon only. To work around this, see..."

File: content/docs/using-appsody/deploying.md [hunk 3]
```diff
@@ -9,55 +9,59 @@ The Appsody CLI provides the [appsody deploy](/docs/cli-commands/#appsody-deploy
...
+
 These deployment options are covered in more detail in the following sections.
 
-## Deploying your app directly to a Kubernetes cluster
+## Deploying your application to a Kubernetes cluster
 There are many options to deploy your Appsody applications to a Kubernetes cluster. The best approach depends on the specific scenario:
-- If you are testing your app on a locally installed cluster, using `appsody deploy` is your best bet
-- If you intend to have your app deployed on a shared cluster for integration testing or production, you are probably going to rely on CI/CD pipelines, and have the app built and deployed from its source.
+- If you intend to have your application deployed on a shared cluster for integration testing or production, you are probably going to rely on CI/CD pipelines, and have the application built and deployed from its source.
```
Comment (helenmasters) [inline content/docs/using-appsody/deploying.md:18]: This sentence is unwieldy... try...

"If you intend to deploy your application on a shared cluster for integration testing or production, you are probably going to rely on CI/CD pipelines. In this case, the application is built and deployed from its source."

File: content/docs/using-appsody/deploying.md [hunk 4]
```diff
@@ -9,55 +9,59 @@ The Appsody CLI provides the [appsody deploy](/docs/cli-commands/#appsody-deploy
...
+- If you are testing your application on a locally installed cluster, you would want to use [appsody deploy](/docs/cli-commands/#appsody-deploy).
 
-The `appsody deploy` command provides a way for you to deploy your application directly to a Kubernetes cluster. The stack contains a deployment manifest that can be consumed by the [Appsody operator](https://operatorhub.io/operator/appsody-operator). `appsody deploy` will install the operator, if necessary, and deploy your application to the cluster using that deployment manifest.
+If you have installed a Kubernetes cluster on your development workstation and want to use your local Docker image cache instead of pushing the image to Docker Hub, make sure you set up your Kubernetes cluster to consume images from the local Docker cache.
```
Comment (helenmasters) [inline content/docs/using-appsody/deploying.md:21]: Can't we shorten this to "If you have a Kubernetes cluster on your development workstation..."

File: content/docs/using-appsody/deploying.md [hunk 5]
```diff
@@ -87,26 +91,12 @@ appsody deploy -t <mynamespace/myrepository[:tag]> --push-url <external-registry
...
-This behavior can be explained by the fact that - if you simply issue `appsody deploy` without explicitly tagging the image - you end up with a deployment manifest (the `app-deploy.yaml` file) that is identical to the one that was used to deploy the application the first time. Therefore, Kubernetes will detect no differences in the deployment yaml, and will do nothing to update your app.
-
-To ensure that the latest version of your app is pushed to the cluster, use the -t flag to add a unique tag every time you redeploy your app. Kubernetes then detects a change in the deployment manifest, and pushes your app to the cluster again. For example: appsody deploy -t dev.local/my-image:0.x, where x is a number that you increment every time you redeploy.
-## Deploying your app through a Tekton pipeline
+## Deploying your application through a Tekton pipeline
 
 > This deployment option is under development
 
-Most likely, the deployment of apps created with the Appsody CLI is going to occur through the invocation of a CI/CD build pipeline.
+Most likely, the deployment of applications created with the Appsody CLI is going to occur through the invocation of a CI/CD build pipeline.
```
Comment (helenmasters) [inline content/docs/using-appsody/deploying.md:98]: ...applications that are created with the Appsody CLI...

File: content/docs/using-appsody/deploying.md [hunk 6]
```diff
@@ -9,55 +9,59 @@ The Appsody CLI provides the [appsody deploy](/docs/cli-commands/#appsody-deploy
...
+Some users have noticed that their code changes do not seem to be published to the target Kubernetes cluster after an initial deployment of the Appsody project through `appsody deploy`.
+
+If you issue `appsody deploy` without explicitly tagging the image, you end up with an identical deployment manifest (`app-deploy.yaml` file) to the one that was initially used to deploy the application. Therefore, Kubernetes will detect no differences in the deployment manifest, and will not update your application.
+
+To ensure the latest version of your application is pushed to the cluster, use the `-t` flag to add a unique tag every time you redeploy your application. Kubernetes then detects a change in the deployment manifest, and pushes your application to the cluster again. For example: `appsody deploy -t dev.local/my-image:0.x`, where x is a number that you increment every time you redeploy.
+
+> If you are running multiple Appsody projects on your workstation, you can use the `appsody deploy` and `appsody operator` commands to get them deployed to a Kubernetes cluster. However, do not run these commands concurrently as they create temporary files that may lead to conflicts.
```
Comment (helenmasters) [inline content/docs/using-appsody/deploying.md:42]: Slight rewording...

"If you are running multiple Appsody projects on your workstation, you can use the `appsody deploy` and `appsody operator` commands to deploy them to a Kubernetes cluster. However, do not run these commands concurrently as they create temporary files that might lead to conflicts."

File: content/docs/faq.md (hunk 1 above)
Comment (skoh7645) [reply content/docs/faq.md:166]: I pasted this directly and it looks much better on the website now! thanks :)

File: content/docs/using-appsody/deploying.md (hunk 4 above)
Comment (skoh7645) [reply content/docs/using-appsody/deploying.md:21]: This is the best link i can find: https://kubernetes.io/docs/concepts/containers/images/#pre-pulled-images

File: content/docs/faq.md [hunk 7]
```diff
@@ -159,4 +159,15 @@ The `project.yaml` file is validated when a new Appsody project is initialized.
 * moved to a different directory
 * are being initialized into the same directory as a previous project
 
-Their entry is removed from the `project.yaml` file and the associated volumes are deleted. This validation ensures that volumes exist for current projects only and that volumes are not reused between different projects of the same name.
\ No newline at end of file
+Their entry is removed from the `project.yaml` file and the associated volumes are deleted. This validation ensures that volumes exist for current projects only and that volumes are not reused between different projects of the same name.
+
+## 12. How do I deploy my application to a local Minikube cluster?
```
Comment (neeraj-laad) [inline content/docs/faq.md:164]: I don't think this title is right. AFAIK, appsody can deploy to minikube fine - no special config should be needed. Please can you confirm this once?

Comment (neeraj-laad) [reply content/docs/faq.md:164]: The intent was to document how to use Docker daemon from minikube vs Docker on host.

**Comment by skoh7645** [comment]
For the FAQ section, `appsody init` is the only command that still works fine with minikube. I'm not sure if we need to document that or if we should just say generally that appsody commands won't work here

File: content/docs/installing/installing-appsody.md [hunk 8]
```diff
@@ -17,7 +17,11 @@ Appsody supports the following operating systems on x86 hardware:
 
 ## Requirements
 
-To use all of Appsody's functions, you need to install [Docker](https://docs.docker.com/get-started/) and start the Docker daemon on your system. In environments where Docker is not available, a subset of Appsody commands can be used with Buildah as detailed in the [FAQ](/docs/faq#9-can-i-use-appsody-without-docker). Prerequisites specific to each operating system are provided within the links in the following section.
+To use all of Appsody's functions, you need to install [Docker](https://docs.docker.com/get-started/) and start the Docker daemon on your system. 
+
+> Currently, Appsody supports using a local Docker daemon only. See the [FAQ](/docs/faq/#13-can-i-use-the-minikube-docker-daemon-for-development).
+
+In environments where Docker is not available, a subset of Appsody commands can be used with Buildah as detailed in the [FAQ](/docs/faq#9-can-i-use-appsody-without-docker). Prerequisites specific to each operating system are provided within the links in the following section.
```
Comment (CameronMcWilliam) [inline content/docs/installing/installing-appsody.md:24]: ".... of the following section"
